### PR TITLE
Fix query log on mobile

### PR DESCRIPTION
**Pull request by Mcat12**
**By submitting this pull request, I confirm the following (please check boxes, eg [X] - no spaces) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

10

---

Old:
screenshot from 2017-09-24 11-49-43

New:
screenshot from 2017-09-24 12-01-23

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

**Comment by DL6ER** [comment]
Not sure why it doesn't seem to work for me. Not actually tested on a mobile device, but with Chrome device toolbar. Tried refreshing + force refresh and I actually see the added `"scrollX": "100%",` line in the source.

screenshot at 2017-09-24 23-33-23

**Comment by Th3M3** [comment]
@Mcat12 As I see, the main reason of malformed table is because of the css-rule
```
#all-queries {
 table-layout: fixed;
}
```
The `table-layout` should not be set to `fixed` if the table has alternating content.
In CSS it's also possible to control the column's width if the table is not set to fixed.

Furthermore the `<div class="table-responsive">` should be moved directly in front of `<table id="all-queries"...>`

Like this:
queries

Result:
queries

**Comment by WaLLy3K** [comment]
The table is looking fantastic now, thank you very much for that input @Th3M3!

Two things:
* Can we add `.dataTables_scroll { -webkit-overflow-scrolling: touch; }` to `/var/www/html/admin/style/pihole.css`? [It doesn't work on all devices](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-overflow-scrolling) (which is why I haven't just modified the file on this branch myself), but (IMHO) *hugely* improves the user experience on iOS/Mac touch input devices. I've tested it locally, and it works as expected
* Can we fix the pagination to not overflow on smaller screens? Examples can be seen above in this PR

**Comment by Mcat12** [comment]
> Can we fix the pagination to not overflow on smaller screens? Examples can be seen above in this PR

I'm not sure if that's easy or possible to do. The overflow that is shown seems to not be soo bad, as the previous button is then right next to the next button.

**Comment by WaLLy3K** [comment]
Issue only occurs on `fix/mobile-queries`
img_0718

**Comment by WaLLy3K** [comment]
The table is looking beautiful now! :grinning: Last thing: is it possible to fix the [pagination to not overflow](https://user-images.githubusercontent.com/3049142/30789342-3061f99c-a1e8-11e7-941c-dfc139261a16.png)?

**Comment by Mcat12** [comment]
Would giving it a horizontal scroll be good?

**Comment by WaLLy3K** [comment]
I'm of a mixed mind on this, so hear me out: For mobile devices, ultimately, there should be no padding - the white section (with the scrollable table) should fill the entire screen to ensure the maximum amount of space for small devices. That should give room to remove one pagination ([5] perhaps?) and have it all fit on one row.

That's a bit outside the scope of this PR though, so the option to scroll the pagination is the next best fix.

**Comment by Th3M3** [comment]
@Mcat12 @WaLLy3K 
I think you need to give it a horizontal scroll if you want to fit all in one row. 
To save space, you can also remove the button text (Whitelist/Blacklist) and leave only the Icon when shown on a mobile screen. But remember that Domain can be something short or very long too and should have enough space.

If you don't want the horizontal scroll, its also possible ti set fixed line-breaks to the table with css 
(Example by using the css code below):
queries_braked

<pre>@media (max-width: 767px) {

	#all-queries td,
	#all-queries th {
		display: block;
		width: 100% !important;
		box-sizing: border-box;
	}

	#all-queries td:nth-of-type(1),
	#all-queries th:nth-of-type(1),
	#all-queries td:nth-of-type(2),
	#all-queries th:nth-of-type(2),
	#all-queries td:nth-of-type(4),
	#all-queries th:nth-of-type(4),
	#all-queries td:nth-of-type(5),
	#all-queries th:nth-of-type(5) {
		display: inline-block;
		width: 50% !important;
	}

	#all-queries td:nth-of-type(2),
	#all-queries td:nth-of-type(5),
	#all-queries td:nth-of-type(6) {
		text-align: right;
	}

	#all-queries td:nth-of-type(1),
	#all-queries td:nth-of-type(4) {
		border-right-color: transparent;
	}

	#all-queries tr:first-child td:nth-of-type(1),
	#all-queries tr:first-child td:nth-of-type(2),
	#all-queries td:nth-of-type(3),
	#all-queries td:nth-of-type(4),
	#all-queries td:nth-of-type(5),
	#all-queries td:nth-of-type(6) {
		border-top-color: transparent;

	}

	#all-queries th {
		border-bottom-width: 1px;
	}

	#all-queries tfoot {
		display: none;
	}
}</pre>

**Comment by WaLLy3K** [comment]
I think having the pagination scroll is a good short-term fix to make the mobile UI usable, and then we can re-look into how to making the content perfectly fit (along the lines that @Th3M3 is suggesting) with another PR.

File: style/pi-hole.css [hunk 1]
```diff
@@ -49,3 +49,9 @@ a.lookatme {
 .dataTables_scroll {
 	-webkit-overflow-scrolling: touch;
 }
+.pagination-scroll {
```
Comment (WaLLy3K) [inline style/pi-hole.css]: Can we also add `-webkit-overflow-scrolling: touch;` to this element too? Aside from that, everything seems to be working as expected!

**Comment by Th3M3** [comment]
I now had a closer look to it but it does not work as expected for me.

The added class `table-responsive` should enclose the whole table, with header and footer, like I showd [here](https://github.com/pi-hole/AdminLTE/pull/586#issuecomment-332578082).

The scrollable pagination breaks if I change the page.
I have tested that on my Mobile (Samsung Browser), in Opera 49.0 and in MS Edge - always while showing all queries (`queries.php?all`).

If it's ok I would do a remake and show that to you, maybe as new PR.

**Comment by WaLLy3K** [comment]
Happy to see if anything we've made can be done better :smile:

**Comment by Mcat12** [comment]
This PR has been superseded by #636

**Pull request closed without merging**